### PR TITLE
feat: schedule report cluster nodes

### PR DESCRIPTION
**Pull request by shmu3l**
image

### CODE REVIEW

File: pkg/clients/cliServiceClient.go [hunk 1]
```diff
@@ -195,3 +195,15 @@ func (c *CliClient) GetVersionRelatedMessages(webhookVersion string) (*VersionRe
...
+type ReportK8sMetadataRequest struct {
+	ClusterUuid   types.UID
+	Token         string
+	NodesCount    int
+	NodesCountErr error
+}
+
+func (c *CliClient) ReportK8sMetadata(request *ReportK8sMetadataRequest) {
+	httpRes, _ := c.httpClient.Request(http.MethodPost, "/cli/workspace/cluster/", request, c.flagsHeaders)
+	fmt.Println(string(httpRes.Body))
```
Comment (TzlilSwimmer123) [inline pkg/clients/cliServiceClient.go:208]: the fmt suppose to be deleted?

File: pkg/k8sMetadataUtil/k8sMetadataUtil.go [hunk 2]
```diff
@@ -2,37 +2,42 @@ package k8sMetadataUtil
...
-
 func InitK8sMetadataUtil() {
 	k8sClient, err := getClientSet()
+
 	if err != nil {
-		errString := fmt.Sprintf("getClientSet error: %s", err.Error())
-		fmt.Println(errString)
-		nodesCount = -1
-		nodesCountErr = errors.New(errString)
+		fmt.Println("Error getting k8s client", err)
```
Comment (TzlilSwimmer123) [inline pkg/k8sMetadataUtil/k8sMetadataUtil.go:24]: the fmt suppose to be removed?

File: pkg/k8sMetadataUtil/k8sMetadataUtil.go [hunk 3]
```diff
@@ -2,37 +2,42 @@ package k8sMetadataUtil
...
-		for {
-			setNodesCount(k8sClient)
-			time.Sleep(time.Hour)
-		}
-	}()
+
+	validator := networkValidator.NewNetworkValidator()
+	cliClient := cliClient.NewCliServiceClient(deploymentConfig.URL, validator)
+	cornJob := cron.New(cron.WithLocation(time.UTC))
+	cornJob.AddFunc("@hourly", func() { sendK8sMetadata(k8sClient, cliClient) })
```
Comment (TzlilSwimmer123) [inline pkg/k8sMetadataUtil/k8sMetadataUtil.go:31]: This is interesting, does cronjob runs in a go routine of it's own?

File: pkg/k8sMetadataUtil/k8sMetadataUtil.go [hunk 4]
```diff
@@ -49,17 +54,23 @@ func getClientSet() (*kubernetes.Clientset, error) {
...
-	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+func getClusterUuid(clientset *kubernetes.Clientset) (types.UID, error) {
+	clusterMetadata, err := clientset.CoreV1().Namespaces().Get(context.TODO(), "kube-system", metav1.GetOptions{})
 	if err != nil {
-		errString := fmt.Sprintf("List error: %s", err.Error())
-		fmt.Println(errString)
-		nodesCount = -1
-		nodesCountErr = errors.New(errString)
-		return
+		return "", err
```
Comment (TzlilSwimmer123) [inline pkg/k8sMetadataUtil/k8sMetadataUtil.go:60]: What is the chance that the cluster uuid will change every hour? maybe we can query it just once

File: pkg/k8sMetadataUtil/k8sMetadataUtil.go [hunk 5]
```diff
@@ -2,37 +2,52 @@ package k8sMetadataUtil
...
-			time.Sleep(time.Hour)
-		}
-	}()
+
+	clusterUuid, _ = getClusterUuid(k8sClient)
+	nodesCount, nodesCountErr = getNodesCount(k8sClient)
+
+	cornJob := cron.New(cron.WithLocation(time.UTC))
+
+	cornJob.AddFunc("@hourly", func() { sendK8sMetadata(nodesCount, nodesCountErr, clusterUuid, cliClient) })
```
Comment (myishay) [inline pkg/k8sMetadataUtil/k8sMetadataUtil.go:40]: this sends the initial `nodesCount` but not updated later

**Review by myishay — APPROVED** [review]

**Pull request merged**